internal/terraform/node_resource_abstract.go

```diff
@@ -204,12 +203,18 @@ func (n *NodeAbstractResource) References() []*addrs.Reference {
 			refs, _ = lang.ReferencesInExpr(addrs.ParseRef, check.ErrorMessage)
 			result = append(result, refs...)
 		}
+	}
 
-		return result
+	for _, importTarget := range n.importTargets {
+		refs, _ := lang.ReferencesInExpr(addrs.ParseRef, importTarget.ID)
+		result = append(result, refs...)
 	}
 
-	// Otherwise, we have no references.
-	return nil
+	if len(result) > 0 {
```

Comment:
might be you could just `return result` here? If it's had nothing added to it then it's just going to be nil anyway.